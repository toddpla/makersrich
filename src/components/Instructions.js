import React, { Component } from 'react';

class Instructions extends Component {

  render() {
    return (
      <div>
        <h2>Instructions</h2>
        <p>
        Welcome to MakersRich!<br/>
        Aim of the game - Find rubies and beans and complete quizes to level<br/>
        BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!
        </p>
        <h3>Controls</h3>
        <p>Use the arrow keys to move around</p>
        <p>Press 'Q' to interact with signs</p>
        <p>Press 'E' to dig</p>
        <p>Press 'I' to open/close your inventory</p>
      </div>
    );
  }

}

export default Instructions;
