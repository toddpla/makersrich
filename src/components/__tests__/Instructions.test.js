import React from 'react';
import { shallow } from 'enzyme'
import Instructions from '../Instructions'

let wrapper

beforeEach(function() {
  wrapper = shallow(
    <Instructions />
  )
});

describe('Instructions', () => {
  it('renders', function() {
    expect(wrapper).toMatchSnapshot()
  });
})
