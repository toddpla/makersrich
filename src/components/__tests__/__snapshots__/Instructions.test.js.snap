// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Instructions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Instructions
        </h2>,
        <p>
          Welcome to MakersRich!
          <br />
          Aim of the game - Find rubies and beans and complete quizes to level
          <br />
          BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!
        </p>,
        <h3>
          Controls
        </h3>,
        <p>
          Use the arrow keys to move around
        </p>,
        <p>
          Press 'Q' to interact with signs
        </p>,
        <p>
          Press 'E' to dig
        </p>,
        <p>
          Press 'I' to open
        </p>,
        <p>
          Press 'Spacebar' to close any pop-ups
        </p>,
      ],
      "id": "instructions-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Instructions",
        },
        "ref": null,
        "rendered": "Instructions",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome to MakersRich!",
            <br />,
            "Aim of the game - Find rubies and beans and complete quizes to level",
            <br />,
            "BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Welcome to MakersRich!",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Aim of the game - Find rubies and beans and complete quizes to level",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Controls",
        },
        "ref": null,
        "rendered": "Controls",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Use the arrow keys to move around",
        },
        "ref": null,
        "rendered": "Use the arrow keys to move around",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press 'Q' to interact with signs",
        },
        "ref": null,
        "rendered": "Press 'Q' to interact with signs",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press 'E' to dig",
        },
        "ref": null,
        "rendered": "Press 'E' to dig",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press 'I' to open",
        },
        "ref": null,
        "rendered": "Press 'I' to open",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Press 'Spacebar' to close any pop-ups",
        },
        "ref": null,
        "rendered": "Press 'Spacebar' to close any pop-ups",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Instructions
          </h2>,
          <p>
            Welcome to MakersRich!
            <br />
            Aim of the game - Find rubies and beans and complete quizes to level
            <br />
            BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!
          </p>,
          <h3>
            Controls
          </h3>,
          <p>
            Use the arrow keys to move around
          </p>,
          <p>
            Press 'Q' to interact with signs
          </p>,
          <p>
            Press 'E' to dig
          </p>,
          <p>
            Press 'I' to open
          </p>,
          <p>
            Press 'Spacebar' to close any pop-ups
          </p>,
        ],
        "id": "instructions-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Instructions",
          },
          "ref": null,
          "rendered": "Instructions",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome to MakersRich!",
              <br />,
              "Aim of the game - Find rubies and beans and complete quizes to level",
              <br />,
              "BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Welcome to MakersRich!",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Aim of the game - Find rubies and beans and complete quizes to level",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "BEWARE! Other players are also competing with you and won't be afraid to challenge you to a showdown of rock, paper scissors an take your stuff!",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Controls",
          },
          "ref": null,
          "rendered": "Controls",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Use the arrow keys to move around",
          },
          "ref": null,
          "rendered": "Use the arrow keys to move around",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press 'Q' to interact with signs",
          },
          "ref": null,
          "rendered": "Press 'Q' to interact with signs",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press 'E' to dig",
          },
          "ref": null,
          "rendered": "Press 'E' to dig",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press 'I' to open",
          },
          "ref": null,
          "rendered": "Press 'I' to open",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Press 'Spacebar' to close any pop-ups",
          },
          "ref": null,
          "rendered": "Press 'Spacebar' to close any pop-ups",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
